src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp

```diff
@@ -484,6 +489,58 @@ static std::vector<Range> getHyperrectangleForRowGroup(const parquet::FileMetaDa
     return hyperrectangle;
 }
 
+ParquetFileMetaDataCache::ParquetFileMetaDataCache(UInt64 max_cache_entries)
+    : CacheBase(max_cache_entries) {}
+
+ParquetFileMetaDataCache *  ParquetFileMetaDataCache::instance(UInt64 max_cache_entries)
+{
+    static ParquetFileMetaDataCache instance(max_cache_entries);
```

Comment:
As I understand, `CacheBase` constuctor first argument is `max_size_in_bytes`, not limit for entries.
https://github.com/ClickHouse/ClickHouse/blob/master/src/Common/CacheBase.h#L47

Reply:
Ouch, I missed this in the first review. I suppose it make sense to create an extra setting for cache byte size then, wdyt?

Reply:
I implemented it, but perhaps we could keep only max_size_bytes instead of max_entries

Reply:
Agree. From my point of view limit in bytes is more predictable that limit in entries.